From fff977164eb14fd35400184b059f4057b02bb76c Mon Sep 17 00:00:00 2001
From: Crestwave <crestwave@users.noreply.github.com>
Date: Mon, 24 Feb 2020 21:29:55 +0800
Subject: openmortal: store config in B_USER_SETTINGS_DIR

Thought I could be clever and just convert HOME to XDG_CONFIG_HOME with
sed, but forgot that it wouldn't be defined when launched from the
Deskbar. :(

diff --git a/src/State.cpp b/src/State.cpp
index a855344..3f1e34c 100644
--- a/src/State.cpp
+++ b/src/State.cpp
@@ -14,6 +14,7 @@
 #include "SDL_keysym.h"
 #include "SDL_mixer.h"
 
+#include <FindDirectory.h>
 #include <string>
 #include <fstream>
 //include <unistd.h>
@@ -44,7 +45,9 @@ std::string GetConfigFilename()
 	}
 	return "c:\\openmortal.ini";	
 #else
-	return std::string(getenv("HOME")) + "/.openmortalrc";
+	char path[B_PATH_NAME_LENGTH];
+	find_directory(B_USER_SETTINGS_DIRECTORY, 0, false, path, sizeof(path));
+	return std::string(path) + "/openmortalrc";
 #endif
 }
 
-- 
2.45.1


From 8d15191d2267c0c6be8c5c2b94dfb27d10c1f2d3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Joachim=20Mairb=C3=B6ck?= <j.mairboeck@gmail.com>
Date: Sun, 9 Jun 2024 15:25:08 +0200
Subject: add detection of freetype via pkg-config


diff --git a/configure.in b/configure.in
index 147d161..0b7facc 100644
--- a/configure.in
+++ b/configure.in
@@ -122,12 +122,21 @@ AC_CHECK_LIB(SDL_net,
 
 dnl Check for Freetype 2 library
 
+PKG_PROG_PKG_CONFIG
+
 FT2_VERSION=2.1.0
+PKG_CHECK_MODULES(
+	[FT2],
+	[freetype2 >= $FT2_VERSION],
+	,
+	[
 AC_CHECK_FT2(
 	$FT2_VERSION,
 	:,
 	AC_MSG_ERROR([*** FreeType 2 version $FT2_VERSION not found!])
 )
+]
+)
 CXXFLAGS="$CXXFLAGS $FT2_CFLAGS"
 LIBS="$LIBS $FT2_LIBS"
 
-- 
2.45.1

